ssl/tls13_enc.c

```diff
@@ -532,6 +538,9 @@ int tls13_change_cipher_state(SSL *s, int which)
                 goto err;
             }
             hashlen = hashlenui;
+#ifndef OPENSSL_NO_QUIC
+            s->quic_len = hashlen;
```

Comment:
s->quic_len also should be set at line 660 for handshake secret.

Reply:
`quic_len` is set once, when the early data secrets are available. L660 is executed later in the handshake, which is after this point.

Reply:
During my test, this line was not hit and s->quic_len was 0 when client handshake read_secret is generated.  If I remember correctly, openssl does not create early secret if no resumption/0RTT attempt is made.

Reply:
I have updated them mechanism for determining the length of the secret.